addOns/oast/src/main/java/org/zaproxy/addon/oast/ExtensionOast.java

```diff
@@ -252,22 +257,16 @@ private void activeScanAlertOastRequestHandler(OastRequest request) {
         try {
             HttpMessage oastReceivedMsg = request.getHistoryReference().getHttpMessage();
             String uri = oastReceivedMsg.getRequestHeader().getURI().toString();
-            if (alertCache.keySet().stream().noneMatch(k -> uri.contains(k.toString()))) {
-                return;
-            }
-            Alert alert = null;
-            for (Object k : alertCache.keySet()) {
-                String key = k.toString();
-                if (uri.contains(key)) {
-                    alert = (Alert) alertCache.get(key);
-                }
-            }
+            Alert alert =
+                    (Alert)
+                            alertCache.keySet().stream()
+                                    .filter(k -> uri.contains(k.toString()))
+                                    .findAny()
+                                    .orElse(null);
```

Comment:
- This code block needs to be synchronized.
- IMO it would be better to keep the old behaviour and warn if the alert was GCed. (Even if before it would be less likely to do it.)

Reply:
That's what a I meant, it's just traversing once now but it still needs to be synchronized (the stream usage needs to be synchronized manually, that's the contract of the returned synchronized map).
If it's about the warn comment, just don't remove the log warn statement.